src/test/ui/const-generics/defaults/trait_objects_fail.stderr

```diff
@@ -2,9 +2,7 @@ error[E0277]: the trait bound `u32: Trait` is not satisfied
   --> $DIR/trait_objects_fail.rs:26:9
    |
 LL |     foo(&10_u32);
-   |     --- ^^^^^^^ the trait `Trait` is not implemented for `u32`
-   |     |
-   |     required by a bound introduced by this call
+   |         ^^^^^^^ the trait `Trait` is not implemented for `u32`
```

Comment:
So the reason for this is that `foo` is not _technically_ the source of the obligation. That is, the predicates of `foo` don't require that `u32: Foo`. The reason that `u32: Trait` is required is because of the implicit cast that happens because the argument is `&dyn Foo`...

I think this can/should be fixed in a follow-up.